src/actions/staking.js

```diff
@@ -1,34 +1,426 @@
+            let totalAvailable = ZERO.clone();
+            let totalPending = ZERO.clone();
+            let total
+            const minimumUnstaked = new BN('100'); // 100 yocto
+
+            try {
+                total = new BN(await validator.contract.get_account_total_balance({ account_id }))
+                if (total.gt(ZERO)) {
+                    validator.staked = await validator.contract.get_account_staked_balance({ account_id })
+                    validator.unclaimed = total.sub(deposited).toString()
+                    validator.unstaked = new BN(await validator.contract.get_account_unstaked_balance({ account_id }))
+                    if (validator.unstaked.gt(minimumUnstaked)) {
+                        const isAvailable = await validator.contract.is_account_unstaked_balance_available({ account_id })
+                        if (isAvailable) {
+                            validator.available = validator.unstaked.toString()
+                            totalAvailable = totalAvailable.add(validator.unstaked)
+                        } else {
+                            validator.pending = validator.unstaked.toString()
+                            totalPending = totalPending.add(validator.unstaked)
+                        }
+                    }
+                } else {
+                    validator.remove = true
+                }
+                totalStaked = totalStaked.add(new BN(validator.staked || ZERO.clone()))
+                totalUnclaimed = totalUnclaimed.add(new BN(validator.unclaimed || ZERO.clone()))
+            } catch (e) {
+                if (e.message.indexOf('cannot find contract code') === -1) {
+                    console.warn('Error getting data for validator', validator.accountId, e)
+                }
+            }
+
+            return {
+                mainAccountId: exAccountId,
+                accountId: account_id,
+                validators: !validator.remove ? [validator] : [],
+                selectedValidator,
+                totalBalance: total.toString(),
+                totalUnstaked: totalUnstaked.toString(),
+                totalStaked: totalStaked.toString(),
+                totalUnclaimed: totalUnclaimed.toString(),
+                totalPending: totalPending.toString(),
+                totalAvailable: totalAvailable.toString(),
+            }
+        },
+        UPDATE_CURRENT: null,
+        GET_LOCKUP: async (accountId) => {
+            let lockupId
+            if (process.env.REACT_APP_USE_TESTINGLOCKUP && accountId.length < 64) {
+                lockupId = `testinglockup.${accountId}`
+            } else {
+                lockupId = getLockupAccountId(accountId)
+            }
+
+            let contract
+            try {
+                await (await new Account(wallet.connection, lockupId)).state()
+                contract = await new Contract(await wallet.getAccountBasic(accountId), lockupId, { ...lockupMethods })
```

Comment:
@marcinbodnar this looks like it wouldn't work with 2FA (because regular `Account`), did you test with 2FA?